### PR TITLE
overlay sys-apps/systemd: Sync with Gentoo, bump to v255.4 and add patches for mutable overlays functionality

### PR DESCRIPTION
**Pull request by krnowak**
Synced with Gentoo, with unnecessary stuff being dropped. This also bumps systemd version to v255.4.

Added the patch implementing the mutable overlays in systemd-sysext - the patch is a merged patch of all the changes I have made and are already in upstream systemd. Not merging the patch would result in having 20 or so extra patches, so it'd be a hindrance. The merged patch implements the minimal mutability and a command-line flag, env var support, ephemeral mutability and some fixes. Changes with documentation and tests are not backported.

CI: https://github.com/flatcar/scripts/pull/1753

- [x] image diff
- [x] changelog

### CODE REVIEW

**Comment by ader1990** [comment]
I think we can also update to the gentoo upstream of 255.4 without any files content changes other than a `mv` on the ebuild file and syncing the upstream gentoo sdk_container/src/third_party/coreos-overlay/sys-apps/systemd/Manifest.

File: sdk_container/src/third_party/coreos-overlay/sys-apps/systemd/systemd-255.4.ebuild [hunk 1]
```diff
@@ -105,6 +107,8 @@ PEFILE_DEPEND='dev-python/pefile[${PYTHON_USEDEP}]'
 # flag). The image stage fails with "Failed to resolve
 # typeattributeset statement at
 # /var/lib/selinux/mcs/tmp/modules/400/ntp/cil:120"
+#
+# Flatcar: Added a dep on sys-apps/kbd. TODO: Why?
```
Comment (krnowak) [inline sdk_container/src/third_party/coreos-overlay/sys-apps/systemd/systemd-255.4.ebuild:111]: @ader1990: Could you give me a pointer about reasons for adding sys-apps/kbd to dependencies? I'll update this comment then.

Comment (ader1990) [reply sdk_container/src/third_party/coreos-overlay/sys-apps/systemd/systemd-255.4.ebuild:111]: sys-apps/kbd contains the loadkeys binary, required by the dracut module https://github.com/dracutdevs/dracut/blob/master/modules.d/10i18n/module-setup.sh, which in turn is required by the dracut module of systemd - vconsole setup https://github.com/dracutdevs/dracut/blob/master/modules.d/00systemd/module-setup.sh#L45 and https://github.com/dracutdevs/dracut/blob/master/modules.d/00systemd/module-setup.sh#L106

Comment (ader1990) [reply sdk_container/src/third_party/coreos-overlay/sys-apps/systemd/systemd-255.4.ebuild:111]: More info here: https://github.com/flatcar/scripts/pull/1679/commits/fcfbf18b8f38f99800d7c81bb68552147cae6f0b

Comment (ader1990) [reply sdk_container/src/third_party/coreos-overlay/sys-apps/systemd/systemd-255.4.ebuild:111]: and here: https://github.com/flatcar/bootengine/pull/87

**Review by ader1990 — COMMENTED** [review]
Without this commit, https://github.com/flatcar/scripts/pull/1679/commits/fcfbf18b8f38f99800d7c81bb68552147cae6f0b, the initrd size will grow by 3MB.

**Comment by krnowak** [comment]
CI passed.

**Review by pothos — APPROVED** [review]

**Pull request merged**